### ISSUE TITLE
New MP Config properties not loaded when new app installed with same name as uninstalled app

### ISSUE THREAD

**Issue by ayoho** [OpenLiberty/open-liberty#14321]
**Describe the bug**
This came up while investigating https://github.com/OpenLiberty/open-liberty/issues/13959. If you have two web applications, both with the same name but different MP Config properties defined, the MP Config runtime does not correctly load the second app's properties. Instead, the properties from the first app will remain in effect.

App 1: "TokenAsCookie"
MP Config properties defined:
- `mp.jwt.verify.issuer=https://server.example.com`
- `mp.jwt.verify.publickey.location=/publicKey.pem`
- `mp.jwt.token.cookie=jwt`

App 2: "TokenAsCookie"
MP Config properties defined:
- `mp.jwt.verify.issuer=https://server.example.com`
- `mp.jwt.verify.publickey.location=/publicKey.pem`
- `mp.jwt.token.header=Cookie`

When the interaction occurs with the first application and the MP Config properties are loaded, I see this in our server logs:
```
[10/2/20, 10:25:13:043 CDT] 00000042 id=00000000 om.ibm.ws.microprofile.config.sources.PropertiesConfigSource > loadProperties Entry  
                                                                                                               wsjar:file:/path/to/build.image/wlp/usr/servers/tckAudNoenv/dropins/TokenAsCookie.war!/META-INF/microprofile-config.properties
[10/2/20, 10:25:13:044 CDT] 00000042 id=00000000 om.ibm.ws.microprofile.config.sources.PropertiesConfigSource < loadProperties Exit  
                                                                                                               {mp.jwt.verify.issuer=https://server.example.com, mp.jwt.token.cookie=jwt, mp.jwt.verify.publickey.location=/publicKey.pem}
```

For the first app, this is correct. But after the new application (with the same name but different MP Config properties) is installed and we perform the same interaction with the app, I see:
```
[10/2/20, 10:25:43:325 CDT] 0000005d id=00000000 om.ibm.ws.microprofile.config.sources.PropertiesConfigSource > loadProperties Entry  
                                                                                                               wsjar:file:/path/to/build.image/wlp/usr/servers/tckAudNoenv/dropins/TokenAsCookie.war!/META-INF/microprofile-config.properties
[10/2/20, 10:25:43:325 CDT] 0000005d id=00000000 om.ibm.ws.microprofile.config.sources.PropertiesConfigSource < loadProperties Exit  
                                                                                                               {mp.jwt.verify.issuer=https://server.example.com, mp.jwt.token.cookie=jwt, mp.jwt.verify.publickey.location=/publicKey.pem}
```

The properties loaded are identical to the first app, despite the second app not having the `mp.jwt.token.cookie` defined and a `mp.jwt.token.header` property defined instead.

**Steps to Reproduce**
1. Configure server with `mpConfig-1.3` or higher. I've observed this with `mpConfig-1.3` and `mpConfig-1.4` alongside the `mpJwt-1.1` and `mpJwt-1.2` features, respectively.
1. Install WAR file that has MP Config properties defined. I've simply been putting the WAR file in `dropins/` for convenience.
1. Interact with the web application so Liberty loads the MP Config properties from the app.
1. Replace the WAR file with a different WAR file with the same name, but with different MP Config properties defined.
1. Interact with the web application again. The MP Config properties from the first application will still be loaded.

**Expected behavior**
The properties for the second app should be loaded when the application is interacted with.

**Diagnostic information:**
```
product = Open Liberty 20.0.0.11 (wlp-1.0.46.202009221019)
wlp.install.dir = /Users/adam/workspaces/libertyGit/open-liberty/dev/build.image/wlp/
java.home = /Library/Java/JavaVirtualMachines/adoptopenjdk-11-openj9.jdk/Contents/Home
java.version = 11.0.7
java.runtime = OpenJDK Runtime Environment (11.0.7+10)
os = Mac OS X (10.15.6; x86_64) (en_US)
```

server.xml:
```XML
<server description="mp_jwt">

   <include location="../fatTestPorts.xml"/>  
   
    <!-- This prevents apps from granting themselves this permission -->
    <javaPermission className="java.security.AllPermission"  name="*" actions="*" />
    
    <featureManager>        
        <feature>componenttest-1.0</feature>
        <feature>jaxrs-2.1</feature>
        <feature>appSecurity-2.0</feature>
        <feature>mpJwt-1.1</feature>
        <feature>localConnector-1.0</feature>
        <feature>arquillian-support-1.0</feature>
    </featureManager>
  
    <mpJwt groupNameAttribute="groups" id="myMpJwt" audiences="s6BhdRkqt3, s6BhdRkqt3.2" clockSkew="10000000000" /> 
  
    <keyStore id="defaultKeyStore" password="keyspass"/> 
</server>
```

I can provide the WAR files I'm using if needed.

**Comment by ayoho** [comment]
FYI: Tom suspects that the problem is here: https://github.com/OpenLiberty/open-liberty/blob/e25ef99ccc22dba8a8f2bb2b5db69f9d8e75c76d/dev/com.ibm.ws.microprofile.config.1.1/src/com/ibm/ws/microprofile/config/impl/AbstractProviderResolver.java#L138

**Comment by ayoho** [comment]
This also appears to be a problem in `mpConfig-2.0`.

**Comment by Azquelt** [comment]
There turned out to be two issues causing this problem:
* JWT was not correctly closing an input stream
  * this was fixed under #14413
* ZipFileHandles were not being refreshed if there was a stream left open
  * we have a more specific issue open for this #14410 

Since the two problems which caused this symptom are either fixed or covered by another issue, I will close this issue.
